packages/jsdoc-plugins/lib/utils/collection.js

```diff
@@ -28,6 +30,11 @@ class Collection {
 		}
 
 		this._data[ name ].push( record );
+		this._allData.push( record );
+
+		if ( record.longname ) {
```

Comment:
Maybe it was a mistake that I made this class a little bit too generic. Now assuming that an object has a property `longname` is confusing. Because the class became here a `DocletCollection` with your changes. I think, that you can rename it, as it's only used for doclets and there's no other use case for that class. Eventually, you could inherit `DocletCollection` from the `Collection`, but as I mention earlier, there's no need for that.

Reply:
Yeah, that came to my mind too. I think it's best to rename it to DocletCollection as you suggest because it's the only use case of the class.